src/main/java/org/testng/internal/Parameters.java

```diff
@@ -52,11 +52,39 @@
       AfterMethod.class
   };
 
-  private static Map<String, List<Class>> mapping = Maps.newHashMap();
+  private static Map<String, List<Class<?>>> mapping = Maps.newHashMap();
+/*
+          +--------------+--------------+---------+--------+----------+-------------+
```

Comment:
Great idea indeed.

Looking more closely to this table, I'm surprised `XmlTest` is allowed in `@BeforeSuite` since that doesn't make sense to me (we haven't entered a `<test>` tag yet). Is this correct?